File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    visited = [False] * n
    answer = 0

    def dfs(current):
        visited[current] = True

        # 현재 컴퓨터와 연결된 컴퓨터들의 연계를 전체 확인
        for next_computer in range(n):
            # 연결되어 있고, 아직 방문하지 않았다면
            if computers[current][next_computer] == 1 and not visited[next_computer]:
                dfs(next_computer)

    for i in range(n):
        # 아직 방문하지 않은 컴퓨터라면 이전에 탐색한 컴퓨터와 네트워크로 연결되지 않은 새 네트워크로 판단
        if not visited[i]:
            answer += 1
            dfs(i)

    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    const edges = Array.from({length: n}, () => []);
    const visited = new Array(n).fill(0);
    let networkCount = 0;
    
    // 간선 정보 만들기
    for(let i = 0; i < n; i++){
        for(let j = 0; j < n; j++){
            if(i === j || computers[i][j] === 0) continue;
            edges[i].push(j);
            edges[j].push(i);
        }
    }
    
    //for문 => dfs (덩어리 개수 찾기)
    for(let i = 0; i < n; i++){
            if(visited[i] === 1) continue;
            visited[i] = 1;
            networkCount++;
            dfs(i);
    }
    
    // dfs 함수
    function dfs (node) {
        for(const next of edges[node]) {
            if(visited[next] === 0){
                visited[next] = 1;
                dfs(next);
            }
        } 
    }
    
    return networkCount;
}